pkg/validate/dynamic/dynamic.go

```diff
@@ -438,22 +424,28 @@ func (c closure) usingCheckAccessV2() (bool, error) {
 		return false, nil
 	}
 
+	actionsToFind := map[string]struct{}{}
 	for _, action := range c.actions {
-		found := false
-		for _, result := range results.Value {
-			if result.ActionId == action {
-				found = true
-				if result.AccessDecision == remotepdp.Allowed {
-					break
-				}
-				return false, nil
+		actionsToFind[action] = struct{}{}
+	}
+	for _, result := range results.Value {
+		_, ok := actionsToFind[result.ActionId]
+		if ok {
+			delete(actionsToFind, result.ActionId)
+			if result.AccessDecision == remotepdp.Allowed {
+				continue
 			}
-		}
-		if !found {
-			c.dv.log.Infof("The result didn't include permission %s", action)
 			return false, nil
 		}
 	}
+	if len(actionsToFind) > 0 {
+		missingActions := []string{}
+		for missingAction := range actionsToFind {
+			missingActions = append(missingActions, missingAction)
+		}
```

Comment:
IMO we don't need to declare a new slice and copy over the `actionsToFind` just to effectively "rename" the slice. We could use `actionsToFind` directly in the log statement.

Reply:
When using `actionsToFind` directly, it gives the error as:-
`The result didn't include permissions map[Microsoft.Network/natGateways/write:{}] for object ID: billyqeillor`

Reply:
It looks like you changed it to align with my suggestion though. How did you end up fixing that error after you encountered it?

Reply:
So the output/error-message in my last comment is the output of a unit-test.
Yes, I changed it to your suggested way. So question here is to keep it like this or revert to the earlier way?
The difference is in the error message format:-
Current:- `The result didn't include permissions map[Microsoft.Network/natGateways/write:{}] for object ID: billyqeillor`
Revert:- `The result didn't include permissions [Microsoft.Network/natGateways/write] for object ID: billyqeillor`

Reply:
Oh... I missed that `actionToFind` is a map and `missingActions` was a slice. I originally thought one slice was just being copied into another slice here.

Now that I properly get what's going on, I'd say to revert this change so we have the nicer error message format.